src/flask/sansio/blueprints.py

```diff
@@ -353,11 +353,11 @@ def register(self, app: App, options: dict) -> None:
 
             if bp_subdomain is None:
                 bp_subdomain = blueprint.subdomain
-
-            if state.subdomain is not None and bp_subdomain is not None:
-                bp_options["subdomain"] = bp_subdomain + "." + state.subdomain
-            elif bp_subdomain is not None:
-                bp_options["subdomain"] = bp_subdomain
+            if bp_subdomain is not None:
+                if state.subdomain is not None:
+                    bp_options["subdomain"] = bp_subdomain + "." + state.subdomain
+                else:
+                    bp_options["subdomain"] = bp_subdomain
```

Comment:
Since the first if statement (line 354) checks if the `bp_subdomain` is None and then sets a value for it, the result of that is that `bp_subdomain` will never be None, starting on line 356. As such, the check on line 356 is redundant and will always be True, so it can be removed entirely.
```suggestion
            if state.subdomain is not None:
                bp_options["subdomain"] = bp_subdomain + "." + state.subdomain
            else:
                bp_options["subdomain"] = bp_subdomain
```